orbit/pkg/update/update.go

```diff
@@ -106,19 +105,25 @@ func NewUpdater(opt Options) (*Updater, error) {
 	}
 
 	tufClient := client.NewClient(opt.LocalStore, remoteStore)
-	var rootKeys []*data.PublicKey
-	if err := json.Unmarshal([]byte(opt.RootKeys), &rootKeys); err != nil {
-		return nil, fmt.Errorf("unmarshal root keys: %w", err)
-	}
 
+	// First try initializing via stored local metadata. Older Orbit packages were distributed with
+	// only the root *key* metadata and not the full root metadata (see
+	// https://github.com/theupdateframework/go-tuf/issues/379), so it's important that we try using
+	// the fully loaded root metadata in the local store before we try using what is provided via
+	// configuration.
 	meta, err := opt.LocalStore.GetMeta()
-	if err != nil || meta["root.json"] == nil {
-		var rootKeys []*data.PublicKey
-		if err := json.Unmarshal([]byte(opt.RootKeys), &rootKeys); err != nil {
-			return nil, fmt.Errorf("unmarshal root keys: %w", err)
+	if err == nil && meta["root.json"] != nil {
+		if err := tufClient.Init(meta["root.json"]); err != nil {
```

Comment:
`main` version that doesn't have a `else` clause is probably a-ok, `tufClient.Init` is only needed when there's no local metadata.